### PR TITLE
Only set host name in recovery when /etc/HOSTNAME (uppercase) exists

### PR DESCRIPTION
**Pull request by ProBackup-nl**
After merging issue #1286 this enhancement beautifies the recovery environment scripting, to not run unnecessary code for:
1. cases where /etc/HOSTNAME does not exist
2. things that the operating system already has taken care of.

Target: Arch Linux
Hostname set in /etc/hostname (lowercase)
Only that /etc/hostname file present, will already set the hostname in (1) system and (2) shell for Arch Linux, without the need for this script to set any of these 2.

### CODE REVIEW

**Comment by jsmeix** [comment]
@ProBackup-nl 
according to
https://github.com/rear/rear/issues/1258#issuecomment-289167853
and
https://github.com/rear/rear/issues/1258#issuecomment-293558696
I think usr/share/rear/skel/default/etc/scripts/boot
i.e. /etc/scripts/boot in the recovery system is
only run in the recovery system when the
the sysv-compatible init process is used
in the recovery system and I asume that
in your recovery system already systemd is used
so that changes in etc/scripts/boot do not matter
in your case and in general when systemd is used.

If you nevertheless like to enhance etc/scripts/boot:
When I looked at that script during debugging
of https://github.com/rear/rear/issues/1258
I was wondering about why it uses $(uname -n)
to show the HOSTNAME value to the user
and not what it had actually set?

In other words:
After calling
<pre>
hostname "$HOSTNAME"
</pre>
I wonder if then
<pre>
echo Hostname set to $(uname -n)
</pre>
really shows the right output.
On my SLES12 system it works
<pre>
# hostname testy

# hostname
testy

# uname -n
testy
</pre>
but woudn't
<pre>
echo Hostname set to $( hostname )
</pre>
be the more straightforward way to do it?

Simply put:
I do not understand why there is the indirection of using
the "uname -n" output instead of the "hostname" output.

**Comment by jsmeix** [comment]
@gdha 
I would appreciate it if you coud you also have a look here.

**Comment by ProBackup-nl** [comment]
I would suggest to keep `$(uname -n)`,  to rule out edge cases where anything goes wrong, that the actually outputs the value of the current hostname setting.

And I think checking for the existence of /etc/HOSTNAME is better then assuming that it is there.

**Comment by gdha** [comment]
See also #1316

**Comment by jsmeix** [comment]
I like the simple way as in https://github.com/rear/rear/pull/1316
to have two files /etc/hostname and /etc/HOSTNAME with
same content in the ReaR recovery system.
Perhaps we should even have those two same files
in any case in the ReaR recovery system?

**Comment by gdha** [comment]
@ProBackup-nl  @jsmeix With pull request e27f27e _100_hostname.sh: Update comment_ being committed we can close this pull request in my view.

**Comment by jsmeix** [comment]
Since https://github.com/rear/rear/pull/1316 is merged
both /etc/hostname and /etc/HOSTNAME exists in any case
so that this pull request in its current form with
<pre>
if [[ -e /etc/HOSTNAME ]] ; then
 ...
</pre>
does no longer make sense.
